content/docs/user-guide/assets/asset-processor/debugging.md

```diff
@@ -6,9 +6,259 @@ weight: 500
+
+## What are common Asset Processor related problems, and paths to address those problems?
+
+### Non Deterministic Product Assets - Different Product Assets with No Source Changes
+
+#### Situation
+With no changes to the code for a builder, and no changes to a source asset, when that job is run multiple times, the contents of product assets end up different than the previous run.
+
+#### Effect
+More assets than expected will show up when attempting to generate Delta Asset Bundles to create content patches for your game, causing your players to download more content than necessary.
+
+#### Common Causes
+Extraneous information is included in the product asset, such as a timestamp. Logic in the builder is non-deterministic.
+
+#### Solutions
+In most cases this will require an engineer to make changes to the internal builder logic to stabilize the product output.
+
+#### Debugging
+Your team will most likely encounter this issue when generating [Asset Bundles](/packaging/asset-bundler/), which is the primary system non-deterministic Product Assets impacts. If a Product Asset shows up in a bundle that you didn't expect to be there because the Source Asset and associated Asset Builder did not change, then debugging tends to start by working backward from there.
+
+Non-deterministic behavior in Product Assets will be introduced by the asset builder that created this product asset, so to identify the cause you'll want to look for this product asset in the Asset Processor's asset tab, or directly in the Asset Database using an external database browsing tool. From there you can find the job that created that product asset. Once you identify the job that created this product asset, you can then find the builder that produces that job, and go to the Process Jobs function for that builder, to find where it outputs the product asset. Working backward from here, you should be able to find the cause of the Product Asset changing by exploring the code. You can also try [debugging the Asset Builder](#DebugAssetBuilders) processing that job, to step through.
+
+To help identify the block of logic in the Product Asset that is changing each time it is processed, you may be able to use a standard file diffing tool using two generated product assets. You can quickly re-generate the product asset by right clicking it in the Asset Processor's asset tab and selecting reprocess source asset.
+
+Another debugging option is setting up automation to catch this early. If you have an automated system that processes assets for your project, you can expand this automation to also compare iterative asset processing to a clean cache asset process and the previous day's assets. Doing this will let you compare the hashes of Product Assets, and find any where the hash has changed when it should not have.
+
+### Same Product Asset generated with different Sub ID
+<a name="ProductAssetSubId"></a>
```

Comment:
Please change all HTML `<a>`  links in this PR to Netlify links. Using Netlify links on the same line as the header will make scrolling to this link work properly.
```suggestion
### Same Product Asset generated with different Sub ID {#ProductAssetSubId}
```

Reply:
Changed all of them.